### ISSUE TITLE
[Bug]: 'dict object' has no attribute 'raw_code' - incompatibility with other packages

### ISSUE THREAD

**Issue by whiteh** [brooklyn-data/dbt_artifacts#414]
## Overview

We are seeing some incompatibilities with testing modules that rely on the raw_code attribute.  It looks like this is popped from the models during upload through dbt_artifacts.

## How to reproduce

- dbt v1.5.4
- dbt_artifacts (v2.6.1) 
- [dbt-unit-testing](https://github.com/EqualExperts/dbt-unit-testing) (v0.3.2)
- dbt-databricks (v1.5.5)
- `on-run-end`:
  - "{{ dbt_artifacts.upload_results(results) }}"
- threads: 1   - seems this is a race condition, so easier to reproduce with a single thread

1. Run `dbt compile`
2. All dbt_artifacts uploads work fine
3. Receive error `Compilation Error in test {test_name} ({test_path}) 'dict object' has no attribute 'raw_code' <stack trace>
4. This property is popped from each uploaded model in dbt_artifacts [here](https://github.com/brooklyn-data/dbt_artifacts/blob/4220a9cea519c98a7d23ec8396d661c6fc8880da/macros/upload_individual_datasets/upload_models.sql#L27)

## Expected behaviour

1. Other processes subsequent to dbt_artifacts upload should finish successfully.
2. dbt_artifacts should not edit the model object

## Environment

- dbt v1.5.4
- dbt_artifacts (v2.6.1) 
- [dbt-unit-testing](https://github.com/EqualExperts/dbt-unit-testing) (v0.3.2)
- dbt-databricks (v1.5.5)

Please paste the contents of your `packages.yml` file here:

```yml
packages:
  packages:
  - package: brooklyn-data/dbt_artifacts
    version: 2.6.1
  - git: "https://github.com/EqualExperts/dbt-unit-testing"
    revision: v0.3.2
```

**Comment by whiteh** [comment]
I have a fix for this in our code base - I'll open a PR shortly.

**Comment by jared-rimmer** [comment]
Resolved in #415 and will be in the next release. Thanks for the contribution @whiteh
